torch/nn/_functions/vision.py

```diff
@@ -10,16 +10,13 @@ class GridSampler(Function):
 
     @staticmethod
     def _enforce_cudnn(input):
-        if not cudnn.enabled:
-            raise RuntimeError("GridSampler needs CuDNN for processing CUDA inputs,"
-                               " but CuDNN is not enabled")
         assert cudnn.is_acceptable(input)
 
     @staticmethod
     def forward(ctx, input, grid):
         ctx.save_for_backward(input, grid)
         grid_sz = grid.size()
-        if input.is_cuda:
+        if input.is_cuda and cudnn.enabled:
```

Comment:
I think it might be better to remove the `_enforce_cudnn` method, and instead check for `cudnn.is_acceptable(input)` here instead of checking for both conditions, as [`cudnn.is_acceptable` already perform those checks for you](https://github.com/pytorch/pytorch/blob/master/torch/backends/cudnn/__init__.py#L37-L56)